Btrixjobs Operator

leepro opened the pull request:
## Context 

* 1st phase (the second diagram in https://github.com/webrecorder/browsertrix-cloud/issues/550#issuecomment-1440685194)
* The sequence of the workflow
 
<img width="692" alt="Screenshot 2023-03-06 at 3 27 59 PM" src="https://user-images.githubusercontent.com/101293/223279977-26c18444-c0de-41a1-bef6-2c4380f1b611.png">

* Add CRDs for `btrixjobs`
* Add Role/RoleBinding for `btrixjobs` in `btrix.cloud` apiGroup
* Add `operator.py` as a webhook handler for Metacontroller
* Add helm chart installation in CI/CD test pipeline
* Add Helm charts (metacontroller + composite controller + CRD)
* Add doc

## Run/Test

```helm upgrade --install -f ./chart/admin/operator/values.yaml btrix-operator ./chart/admin/operator```

leepro committed:
init operator

leepro committed:
init files

leepro committed:
operator and rename crd

leepro committed:
update

leepro committed:
reformat

leepro committed:
install operator

leepro committed:
add roles for btrix.cloud apiGroup

leepro committed:
convert it helm chart

leepro committed:
update doc

leepro committed:
let it use latest version

ikreymer commented:
Let's combine the metacontroller to be part of the main chart (which will depend on metacontroller), rather than its own subchart

leepro committed:
update helm chart

leepro committed:
remove the separate helm chart for operator

leepro committed:
Merge branch 'main' into btrix-operator

leepro committed:
Merge branch 'main' into btrix-operator

ikreymer committed:
Merge branch 'main' into btrix-operator

ikreymer commented:
Hm, does it still make sense to have a separate admin/operator chart, rather than just moving that into the chart/ and adding metacontroller dependency there? Since this won't be an optional package, that seems clearer.

Also, should probably make it run the metacontroller endpoints api on a different port than public api, if we're keeping it in the same image - that could probably go either way.
We could have a separate deployment that's separate from backend, or just make it be part of backend that runs two FastAPI servers, or a separate deployment for public api and btrix operator, from mostly same codebase..

I think we'll also be getting rid of all of the *_job.py entrypoints, which will simplify things a bit!

ikreymer committed:
operator fixes:
- run on separate port
- btrix-job: add missing 'rev' field
- add 'main_op' file

leepro commented:
> Hm, does it still make sense to have a separate admin/operator chart, rather than just moving that into the chart/ and adding metacontroller dependency there? Since this won't be an optional package, that seems clearer.

Agree.

> 
> Also, should probably make it run the metacontroller endpoints api on a different port than public api, if we're keeping it in the same image - that could probably go either way. We could have a separate deployment that's separate from backend, or just make it be part of backend that runs two FastAPI servers, or a separate deployment for public api and btrix operator, from mostly same codebase..
> 
> I think we'll also be getting rid of all of the *_job.py entrypoints, which will simplify things a bit!

Yes, As we discussed before, this operator will be taken out of the current backend.

leepro committed:
merge the operator into main chart

leepro committed:
Merge branch 'btrix-operator' of https://github.com/webrecorder/browsertrix-cloud into btrix-operator

leepro committed:
fix
